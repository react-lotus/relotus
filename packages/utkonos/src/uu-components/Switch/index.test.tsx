/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import type { Ref } from 'react';
import { createRef } from 'react';
import { Switch, SwitchProps } from '.';

const SwitchWrapper = (props: Partial<SwitchProps> & { customRef?: Ref<HTMLInputElement> }) => {
  const { customRef, ...rest } = props;
  return <Switch ref={customRef} {...rest} />;
};

describe('Switch', () => {
  describe('snapshot', () => {
    it('default', () => {
      const component = <Switch />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('checked', () => {
      const component = <Switch defaultChecked />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled', () => {
      const component = <Switch disabled />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('disabled checked', () => {
      const component = <Switch disabled defaultChecked />;

      const tree = renderer.create(component).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('component', () => {
    it('should properly pass ref to the Switch element', () => {
      const ref = createRef<HTMLInputElement>();
      const { getByRole } = render(<SwitchWrapper customRef={ref} />);
      ref.current?.focus();
      expect(getByRole('checkbox')).toHaveFocus();
    });
  });
});
