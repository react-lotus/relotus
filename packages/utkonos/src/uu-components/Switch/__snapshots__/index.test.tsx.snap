// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot checked 1`] = `
<label
  className="uu-switch"
  htmlFor="nanoid"
>
  <input
    className="uu-switchInput"
    defaultChecked={true}
    id="nanoid"
    type="checkbox"
  />
  <span
    className="uu-switchTrack"
  />
  <span
    className="uu-switchThumb"
  />
</label>
`;

exports[`Switch snapshot default 1`] = `
<label
  className="uu-switch"
  htmlFor="nanoid"
>
  <input
    className="uu-switchInput"
    id="nanoid"
    type="checkbox"
  />
  <span
    className="uu-switchTrack"
  />
  <span
    className="uu-switchThumb"
  />
</label>
`;

exports[`Switch snapshot disabled 1`] = `
<label
  className="uu-switch uu-switch_disabled"
  htmlFor="nanoid"
>
  <input
    className="uu-switchInput"
    disabled={true}
    id="nanoid"
    type="checkbox"
  />
  <span
    className="uu-switchTrack uu-switchTrack_disabled"
  />
  <span
    className="uu-switchThumb uu-switchThumb_disabled"
  />
</label>
`;

exports[`Switch snapshot disabled checked 1`] = `
<label
  className="uu-switch uu-switch_disabled"
  htmlFor="nanoid"
>
  <input
    className="uu-switchInput"
    defaultChecked={true}
    disabled={true}
    id="nanoid"
    type="checkbox"
  />
  <span
    className="uu-switchTrack uu-switchTrack_disabled"
  />
  <span
    className="uu-switchThumb uu-switchThumb_disabled"
  />
</label>
`;
